EIPS/eip-1238.md

```diff
@@ -0,0 +1,231 @@
+
+```solidity
+interface IERC1238Expirable is IERC1238 {
+    /**
+     * @dev Returns the expiry date for tokens with a given `id`.
+     */
+    function expiryDate(uint256 id) external view returns (uint256);
+
+    /**
+     * @dev Returns whether tokens are expired by comparing their expiry date with `block.timestamp`.
+     */
+    function isExpired(uint256 id) external view returns (bool);
+
+    /**
+     * @dev Sets the expiry date for the tokens with id `id`.
+     */
+    function setExpiryDate(uint256 id, uint256 date) external;
+
+    /**
+     * @dev [Batched] version of {setExpiryDate}.
+     */
+    function setBatchExpiryDates(uint256[] memory ids, uint256[] memory dates) external;
+}
+```
+
+The **Holdable extension** is OPTIONAL for smart contracts that implement this EIP. This extension enables token owners to let their tokens be held by another address, including "staking" them in a smart contract.
+Since NTTs are non-transferable, the usual way of staking tokens by transferring ownership over them is not possible. In order to make staking possible, this extension introduces the notion of token holders, in addition to token owners:
+
+- When minting tokens to Alice's address A, address A is the token owner but also the initial token holder. Alice can then call the `entrust` function to entrust address B with her tokens. Address B becomes the token holder while Alice remains the token owner.
+- Only one address can hold specific tokens at a time.
+- A token holder can transfer tokens to another holder. Still, the important point is that this does not change ownership of the tokens.
+- Because token holders can “transfer” tokens they hold (via the `entrust` function) to any address they choose, they can also send them to the zero address and have it hold these tokens forever. So token owners are effectively putting their tokens at stake with this mechanism.
+
+```solidity
+interface IERC1238Holdable is IERC1238 {
+    /**
+     * @dev Event emitted when `from` entrusts `to` with `amount` of tokens with token `id`.
+     */
+    event Entrust(address from, address to, uint256 indexed id, uint256 amount);
+
+    /**
+     * @dev Event emitted when tokens are burnt and the holder fails to acknowledge the burn.
+     */
+    event BurnAcknowledgmentFailed(address holder, address burner, address from, uint256 indexed id, uint256 amount);
+
+    /**
+     * @dev Returns the balance of a token holder for a given `id`.
+     */
+    function heldBalance(address holder, uint256 id) external view returns (uint256);
+
+    /**
+     * @dev Lets sender entrusts `to` with `amount`
+     * of tokens, which gets transferred between their respective balances
+     * of tokens held.
+     */
+    function entrust(
+        address to,
+        uint256 id,
+        uint256 amount
+    ) external;
+}
+```
+
+Smart contracts holding tokens MUST implement the following interface:
```

Comment:
This standard really can't place requirements on smart contracts that don't claim to implement it. I'd reword this to something like:

> Smart contracts that are intended to hold NTTs SHOULD implement the following interface to receive burn notifications: